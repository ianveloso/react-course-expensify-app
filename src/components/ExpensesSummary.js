import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expensesCount = 0, expensesTotal = 0}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    
    return (
        <div>
            <h1>Viewing {expensesCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters); 

    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary); 
