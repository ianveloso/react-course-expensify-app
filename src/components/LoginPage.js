import React from 'react';
import { connect } from 'react-redux'; 
import { startLogin } from '../actions/auth'; 

export const LoginPage = ({ startLogin }) => (
    <div>
        <button onClick={startLogin}>Login</button>
    </div>    
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => {
        console.log('dispatching login');
        dispatch(startLogin())
    }
})

export default connect(undefined, mapDispatchToProps)(LoginPage); 
