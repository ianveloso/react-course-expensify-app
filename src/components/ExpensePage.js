import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpensePage = () => (
    <div>
        This is from my dashboard component
        <ExpenseList />
        <ExpenseListFilters />
        <ExpensesSummary />
    </div>
);

export default ExpensePage;