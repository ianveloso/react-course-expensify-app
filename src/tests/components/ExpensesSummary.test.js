import React from 'react'; 
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render expenses summary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render expenses summary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={23523124124} />);
    expect(wrapper).toMatchSnapshot();
});