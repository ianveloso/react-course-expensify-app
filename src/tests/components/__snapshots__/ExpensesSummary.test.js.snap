// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render expenses summary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $235,231,241.24
  </h1>
</div>
`;
